src/object/custom_particle_system.cpp

```diff
@@ -1241,4 +1246,1024 @@ CustomParticleSystem::ease_value(float* value, float target, float time, easing
   script_easings.push_back(req);
 }
 
+void
+CustomParticleSystem::spawn_particles(int amount, bool instantly)
+{
+  if (instantly)
+  {
+    for (int i  = 0; i < amount; i++)
+    {
+      spawn_particles(0.f);
+    }
+  }
+  else
+  {
+    // TODO: Implement delayed spawn mode for scripting.
+    log_warning << "Delayed spawn mode is not yet implemented for scripting." << std::endl;
+  }
+}
+
+// =============================================================================
+// ============================   ATTRIBUTES   =================================
+// =============================================================================
+
+int
+CustomParticleSystem::get_max_amount() const
+{
+  return m_max_amount;
+}
+
+void
+CustomParticleSystem::set_max_amount(int amount)
+{
+  m_max_amount = amount;
+}
+
+bool
+CustomParticleSystem::get_cover_screen() const
+{
+  return m_cover_screen;
+}
+
+void
+CustomParticleSystem::set_cover_screen(bool cover)
+{
+  m_cover_screen = cover;
+}
+
+std::string
+CustomParticleSystem::get_birth_mode() const
+{
```

Comment:
I would maybe introduce a `using FadeMode = CustomParticleSystem::FadeMode` here or at the start.